pyhik/hikvision.py

```diff
@@ -677,3 +677,149 @@ def update_attributes(self, event, channel, attr):
         except KeyError:
             _LOGGING.debug('Error updating attributes for: (%s, %s)',
                            event, channel)
+
+    def inject_events(self, events):
+        """Inject discovered events into the camera's event_states.
+
+        This allows the camera to track events that wouldn't normally be
+        detected, such as those from NVRs with non-standard notification
+        methods.
+
+        Args:
+            events: Dict mapping event type names to lists of channel numbers.
+        """
+        for event_name, channels in events.items():
+            for channel in channels:
+                # Only add if not already present
+                if event_name not in self.event_states:
+                    self.event_states[event_name] = []
+
+                # Check if this channel is already tracked
+                channel_exists = any(
+                    sensor[1] == channel for sensor in self.event_states[event_name]
+                )
+                if not channel_exists:
+                    # Add the event state: [is_active, channel, count, last_update_time]
+                    self.event_states[event_name].append(
+                        [False, channel, 0, datetime.datetime.now()]
+                    )
+
+
+def get_nvr_events(host, port=DEFAULT_PORT, usr=None, pwd=None, verify_ssl=True):
```

Comment:
This function seems unnecessary as it duplicates the main event fetcher just to add more filter options.  It would be better to modify the existing event function to accept a dict with the default "center" and "http" being the current functionality, but allow for someone to pass in others as needed.

Let me know if you have a use case where that wouldn't work.